File: iotcloud-restapi-java/Core/src/com/qcloud/restapi/iotcore/response/ListProductsResponse.java
package com.qcloud.restapi.iotcore.response;

import com.google.gson.*;

import java.util.ArrayList;
import java.util.List;

public class ListProductsResponse extends Response {
    public class ProductInfo {
        public String productName = "";
        public String productID = "";
        public String productDescription = "";
        public String encryptionType = "";
        public String region = "";
        public String creationDate = "";

        public ProductInfo() {

        }
    }

    /**
     * 解析JSON字串并构造ListProductsResponse对象
     *
     * @param response 从服务端返回的应答JSON字串
     */
    public ListProductsResponse(String response) {
        super(response);
    }

    public List <ProductInfo> getProducts() {
        List <ProductInfo> list = new ArrayList<>();

        try {
            JsonArray arr = jsonObj.getAsJsonArray("products");
            for (int i = 0; i < arr.size(); i++) {
                ProductInfo info = new ProductInfo();

                info.productName = arr.get(i).getAsJsonObject().get("productName").getAsString();
                info.productID = arr.get(i).getAsJsonObject().get("productID").getAsString();

                info.productDescription = arr.get(i).getAsJsonObject().get("productProperties").getAsJsonObject().get("productDescription").getAsString();
                info.encryptionType = arr.get(i).getAsJsonObject().get("productProperties").getAsJsonObject().get("encryptionType").getAsString();
                info.region = arr.get(i).getAsJsonObject().get("productProperties").getAsJsonObject().get("region").getAsString();

                info.creationDate = arr.get(i).getAsJsonObject().get("productMetadata").getAsJsonObject().get("creationDate").getAsString();

                list.add(info);
            }
        }catch (Exception e) {
        }

        return list;
    }
}

File: iotcloud-restapi-php/Core/src/IoTCloud/Response/ListProductsResponse.php
<?php

namespace TXIoTCloud\Response;

require_once "BaseResponse.php";

class ListProductsResponse extends BaseResponse
{
    /**
     * 产品总数
     */
    private $totalCnt;

    /**
     * 产品对象的数组，数组元素为ProductInfo实例
     */
    private $products;

    public function __construct()
    {
    }

    /**
     * 获取产品总数
     */
    public function getTotalCnt()
    {
        return $this->totalCnt;
    }

    /**
     * 设置产品总数
     * @param $totalCnt
     */
    public function setTotalCnt($totalCnt)
    {
        $this->totalCnt = $totalCnt;
    }

    /**
     * 获取产品对象的数组
     */
    public function getProducts()
    {
        return $this->products;
    }

    /**
     * 设置产品对象的数组
     * @param $products
     */
    public function setProducts($products)
    {
        $this->products = $products;
    }


}